### ISSUE TITLE
[java] GuardLogStatement does not recognize when a string is a compile time constant expression

### ISSUE THREAD

**Issue by krallus** [pmd/pmd#3283]
**Affects PMD Version: 6.33.0 (via Eclipse PMD For Eclipse 4.22.0)** 

**Rule: [GuardLogStatement](https://pmd.github.io/pmd-6.33.0/pmd_rules_java_bestpractices.html#guardlogstatement)**

**Description:**

The GuardLogStatement rule is smart enough to know when a guard is needed to avoid work done in the log parameters.  However, it mistakenly identifies concatenation of constant strings as work when these are actually compile time constant expressions.

**Code Sample demonstrating the issue:**

```
final Object param = null;
logger.trace("Some logging string. Contains value: {}", param);
logger.trace("Some long string that has been " + 
         "separated into two constants. Contains value: {}", param);
```

**Expected outcome:**

PMD reports a violation for the second logger call, but that's wrong. That's a false positive.

**Running PMD through:** *Eclipse*

**Comment by krallus** [comment]
I just noticed that this is a duplicate of #957.
